crates/engine/tree/src/persistence.rs

```diff
@@ -244,8 +225,14 @@ where
         while let Ok(action) = self.incoming.recv() {
             match action {
                 PersistenceAction::RemoveBlocksAbove((new_tip_num, sender)) => {
-                    self.remove_blocks_above(new_tip_num).expect("todo: handle errors");
-                    self.remove_static_file_blocks_above(new_tip_num).expect("todo: handle errors");
+                    let provider_rw = self.provider.provider_rw().expect("todo: handle errors");
+                    let sf_provider = self.provider.static_file_provider();
+
+                    self.remove_blocks_above(new_tip_num, &provider_rw, &sf_provider)
+                        .expect("todo: handle errors");
+
+                    provider_rw.commit().expect("todo: handle errors");
+                    sf_provider.commit().expect("todo: handle errors");
```

Comment:
suggestion for followups, it would be nice to have these in separate handler functions like `on_remove_block_above`